runtime/parachains/src/paras_inherent.rs

```diff
@@ -627,31 +637,45 @@ impl<T: Config> Pallet<T> {
 				let freed = collect_all_freed_cores::<T, _>(freed_concluded.iter().cloned());
 
 				<scheduler::Pallet<T>>::clear();
-				<scheduler::Pallet<T>>::schedule(freed, <frame_system::Pallet<T>>::block_number());
+				let now = <frame_system::Pallet<T>>::block_number();
```

Comment:
A tiny optimization to save a read to the storage cache would be to move `now` to a param of `enter_inner` and `create_inherent_inner` so we don't have to call `<frame_system::Pallet<T>>::block_number()` in both

Reply:
Can you create a follow up PR, this is not critical.